instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/handler.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module AwsSdk
+      # Generates Spans for all interactions with AwsSdk
+      class Handler < Seahorse::Client::Handler
+        def call(context)
+          span_name, attributes = calculate_span(context)
+
+          tracer.in_span(span_name, kind: OpenTelemetry::Trace::SpanKind::CLIENT, attributes: attributes) do |span|
+            execute = proc {
+              super(context).tap do |response|
+                if (err = response.error)
+                  span.record_exception(err)
+                  span.status = Trace::Status.error(err)
+                end
+              end
+            }
+
+            if instrumentation_config[:suppress_internal_instrumentation]
```

Comment:
✨ I really like this! 

@open-telemetry/ruby-maintainers I do realize we don't love using `untraced` but I think this would be great to include as an option for other instrumentation where we may have redundant client instrumentation (if we are not doing so already)

Reply:
@arielvalentin 
Agree!
It reduce the amount of redundant spans and making the trace much clearer.

Many nodejs otel instrumentations accept this option, like [opentelemetry-instrumentation-aws-sdk](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-aws-sdk/README.md#aws-sdk-instrumentation-options).
`suppress_internal_instrumentation` option should always be `false` by default to make sure adding new instrumentations won't remove existing spans.